cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/utils/NGSICharsetsTest.java

```diff
@@ -36,237 +36,1616 @@ public NGSICharsetsTest() {
+                + "-------- '=' (internal concatenator) is encoded as \"xffff\" (public concatenator)");
+        String in = "=";
+        String expected = "xffff";
+        String out = NGSICharsets.encodePostgreSQL(in);
+        
+        try {
+            assertEquals(expected, out);
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMySQL]")
+                    + "-  OK  - '" + in + "' has been encoded as '" + expected + "'");
+        } catch (AssertionError e) {
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMySQL]")
+                    + "- FAIL - '" + in + "' has not been encoded as '" + out + "' instead of '" + expected + "'");
+            throw e;
+        } // try catch
+    } // testEncodeMySQLEquals
+    
+    /**
+     * [NGSICharsets.encodeMySQL] -------- A single 'x' is not encoded.
+     */
+    @Test
+    public void testEncodeMySQLSinglex() {
+        System.out.println(getTestTraceHead("[NGSICharsets.encodeMySQL]")
+                + "-------- A single 'x' is not encoded");
+        String in = "x";
+        String expected = "x";
+        String out = NGSICharsets.encodeMySQL(in);
+        
+        try {
+            assertEquals(expected, out);
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMySQL]")
+                    + "-  OK  - '" + in + "' has not been encoded");
+        } catch (AssertionError e) {
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMySQL]")
+                    + "- FAIL - '" + in + "' has encoded as '" + out + "'");
+            throw e;
+        } // try catch
+    } // testEncodeMySQLSinglex
+    
+    /**
+     * [NGSICharsets.encodeMySQL] -------- "xffff" is encoded (escaped) as "xxffff".
+     */
+    @Test
+    public void testEncodeMySQLxffff() {
+        System.out.println(getTestTraceHead("[NGSICharsets.encodeMySQL]")
+                + "-------- \"xffff\" is encoded (escaped) as \"xxffff\"");
+        String in = "xffff";
+        String expected = "xxffff";
+        String out = NGSICharsets.encodeMySQL(in);
+        
+        try {
+            assertEquals(expected, out);
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMySQL]")
+                    + "-  OK  - '" + in + "' has been encoded as '" + expected + "'");
+        } catch (AssertionError e) {
+            System.out.println(getTestTraceHead("[NGSICharsets.encodeMySQL]")
+                    + "- FAIL - '" + in + "' has been encoded as '" + out + "' instead of '" + expected + "'");
+            throw e;
+        } // try catch
+    } // testEncodeMySQLxffff
+    
+    /**
+     * [NGSICharsets.encodeMySQL] -------- Non lower case alphanumerics nor undersocre are encoded.
+     */
+    @Test
+    public void testEncodeMySQLRare() {
+        System.out.println(getTestTraceHead("[NGSICharsets.encodeMySQL]")
+                + "-------- Non lower case alphanumerics and undersocre are encoded");
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1133/commits/3ba31a420f9ddda138c790c11f4d8fdc426ba8f4